pkgs/native_assets_cli/lib/src/model/link_mode_preference.dart

```diff
@@ -2,63 +2,63 @@
 // for details. All rights reserved. Use of this source code is governed by a
 // BSD-style license that can be found in the LICENSE file.
 
-import 'packaging.dart';
+import 'link_mode.dart';
 
-class PackagingPreference {
+class LinkModePreference {
   final String name;
   final String description;
-  final List<Packaging> preferredPackaging;
-  final List<Packaging> potentialPackaging;
+  final List<LinkMode> preferredLinkMode;
```

Comment:
I didn't touch this yet because of :

> LinkingPreference { static, dynamic, any } means we cannot eagerly fail on the build in the package itself returning an error and non-0 exit code. Maybe that's okay, but it would lead to some wasted work if it's a longer build.

Let's solve that question first, then I clean this up.